NLB-7870: Provide guidance on upgrading to NGINX Plus 37.0.4 (#2172)
nginx/documentation: content/nginxaas-azure/known-issues.md

List of known issues in the latest releases of F5 NGINXaaS for Azure.

> ### {{% nginx37-impact %}} NGINX Plus 37.0 behavior impact

Reviewer: This shortcode `{{% nginx37-impact %}}` is causing the build failure. Either remove it or change it to an icon, as shown elsewhere in this file.

Author: Where do I see the error output from the build?  Also, can I create a shortcut to this section to be referenced elsewhere?

Author: Removed it.

> The release of NGINX Plus [37.0](https://docs.nginx.com/nginx/releases/#pls.37.0.4) introduces the following behavioral changes that may affect upstream applications:

Reviewer 2: This should be a ref link

Author: Changed

> - HTTP/1.1 is now the default protocol.

Reviewer 3: it's worth linking to https://blog.nginx.org/blog/keep-alive-to-upstreams-is-now-default-in-nginx-1-29-7

Author: I linked to the 37 release notes at the bottom of this section which also now highlights the changes.  If you think I need both, let me know (I don't want to overload with links).

> - HTTP/1.1 is now the default protocol.

Reviewer 4: ```suggestion
- HTTP/1.1 is now the default protocol for connecting to proxy or upstream servers.
```

> - HTTP/1.1 defaults to keepalive connections.

Reviewer 4: ```suggestion
- HTTP/1.1 defaults to upstream keepalive connections.
```

Reviewer 4: or .. defaults to keeaplive connections between NGINX and upstream server

> - HTTP/1.1 defaults to keepalive connections between NGINX and upstream servers.

Reviewer 4: ```suggestion
- Keepalive connections between NGINX and upstream servers are enabled by default.
```

Author: Done

> - Upstream shared memory zone requires an additional 1KB of memory per upstream server.

Reviewer 4: this should increase shared memory zone sizes by 25-30% (not sure if it's worth mentioning, up to you)

Author: That may be true, and I saw that number mentioned on a MR (if my memory serves me), but I want to keep the wording closer to the official recommendation (https://docs.nginx.com/nginx/releases/#pls.37.0.4)

> To preserve existing behavior, make the follow configuration changes before NGINXaaS is upgraded to NGINX Plus 37.0 (August 17, 2026) on the stable upgrade channel:

Reviewer 4: ```suggestion
To preserve existing behavior, make the following configuration changes before NGINXaaS is upgraded to NGINX Plus 37.0 (August 17, 2026) on the stable upgrade channel:
```

> - Explicitly set the default HTTP version to 1.0:

Reviewer 4: ```suggestion
- Explicitly set the HTTP version to 1.0 when communicating with upstream or proxy servers:
```

> To preserve existing behavior, make the following configuration changes to your NGINX Plus [R36-P8](https://docs.nginx.com/nginx/releases/#r36) configuration before NGINXaaS is upgraded to NGINX Plus 37.0.4 (August 17, 2026) on the stable upgrade channel:

Reviewer 2: This also should be a ref link

Author: Changed

```shell
http {
   proxy_http_version 1.0;
   proxy_set_header Connection "close";
   ...
}
```

> - Set the keepalive directive to zero for all HTTP/1.1 upstream blocks where it has not been explicitly defined:

Reviewer 4: ```suggestion
- Set the [`keepalive`](https://nginx.org/en/docs/http/ngx_http_upstream_module.html#keepalive) directive to zero for all HTTP/1.1 upstream blocks where it has not been explicitly defined:
```

```shell
upstream backend {
   ...
   keepalive 0;
}
```
- Increase the size by approximately 1KB per upstream server, rounding up to leave some headroom:
```shell
upstream backend {
   ...
   zone backend 64k; # Example only. New size depends on previous settings.
}
```

> For further information on the new behavior of NGINX Plus 37.0 see **NGINX Plus PLS.37.0.4.1 LTS** [Upgrade Notes](https://docs.nginx.com/nginx/releases/#pls.37.0.4).

Reviewer 4: ```suggestion
For further information on the new behavior of NGINX Plus 37.0, see **NGINX Plus PLS.37.0.4.1 LTS** [Upgrade Notes](https://docs.nginx.com/nginx/releases/#pls.37.0.4).
```
